pkg/kubectl/cmd/resource/get_test.go

```diff
@@ -980,6 +980,37 @@ func TestGetByFormatForcesFlag(t *testing.T) {
 	}
 }
 
+func TestLabelsColumnWithOutputFlag(t *testing.T) {
+	options := &GetOptions{}
+	testCases := []struct {
+		flags       map[string]string
+		expectedErr string
+	}{
+		{flags: map[string]string{"show-labels": "true"}},
+		{flags: map[string]string{"show-labels": "true", "output": "wide"}},
+		{flags: map[string]string{"show-labels": "true", "output": "yaml"}, expectedErr: "--show-labels option cannot be used with yaml printer"},
+	}
+
+	for _, test := range testCases {
+		f, _, _, _ := cmdtesting.NewAPIFactory()
+
+		buf := bytes.NewBuffer([]byte{})
+		errBuf := bytes.NewBuffer([]byte{})
+
+		cmd := NewCmdGet(f, buf, errBuf)
+		for k, v := range test.flags {
+			cmd.Flags().Lookup(k).Value.Set(v)
+		}
+		errString := ""
```

Comment:
```
err := options.Validate(cmd)
if err != nil && err.Error() != test.expectedErr {
    t.Fatalf("expected errors: %v, but actually got: %v", test.expectedErr, err.Error())
}
if test.expectedErr != "" {
    t.Fatalf("should be error out, but actually not")
}
```
The first check is make sure the Error string is `--show-labels option cannot be used with yaml printer` when you specify both flags.
The second check is also necessary IMO, because what if someone change the functionality and then it doesn't error out, it should throw errors cause we expect it to be error out.

Reply:
@shiywang sorry,I don't get your point. if the function we are going to test doesn't error out, it will meet the condition `err == nil && test.expectedErr != ""` then throw error `"should be error out, but actually not"` Did I misunderstand or miss sth？

Reply:
if the function we are going to test we doesn't expect to error out like those two
```
{flags: map[string]string{"show-labels": "true"}},
{flags: map[string]string{"show-labels": "true", "output": "wide"}},
```
`err == nil` but `test.expectedErr == ""`  so no error

Reply:
yes. According to [function to test](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/resource/get.go#L218) `""` and `wide` are correct, others should error out. So I use the three cases to cover the function test. Is that the right way?